pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -481,6 +481,16 @@ static void validatePartialUpsertStrategies(TableConfig tableConfig, Schema sche
       UpsertConfig.Strategy columnStrategy = entry.getValue();
       Preconditions.checkState(!primaryKeyColumns.contains(column), "Merger cannot be applied to primary key columns");
 
+      if (upsertConfig.getComparisonColumn() != null) {
+        Preconditions.checkState(upsertConfig.getComparisonColumn().compareTo(column) != 0,
+            "Merger cannot be applied to comparison column");
+      }
+
+      if (tableConfig.getValidationConfig().getTimeColumnName() != null) {
```

Comment:
We need to perform this check only when the comparison column is null. Also time column should never be null for upsert table (real-time table)